synapse/handlers/presence.py

```diff
@@ -127,9 +127,11 @@ def __init__(self, hs: "HomeServer"):
         self.state = hs.get_state_handler()
 
         self._federation = None
-        if hs.should_send_federation():
+        if hs.should_send_federation() or not hs.config.worker_app:
             self._federation = hs.get_federation_sender()
```

Comment:
is this bit going away again soon? If so, it's ok. If not, it could do with more comments.

Reply:
Yup, it is.